### ISSUE TITLE
Incomplete install "Insufficient system resources exist to complete the requested service"  on Rpi3B+ for both Win 10 and 11. 

### ISSUE THREAD

**Issue by dgarra** [Botspot/wor-flasher#33]
1) Excellent effort and work here. 

I think this is windows itself, but is there soft threshold on like core count or memory before the windows installer bombs out? This only has 1GB of RAM, so not shocking, but hey, the option is there....

This seems to fail in the same spot on both versions 10 and 11 when selecting the Rpi3B+ and leaving everything default. 

I'm writing to a 29GB SD card. Runs for about an hour before dying citing resources, same failure 5 times in a row. 

```
2021-09-04 04:07:53.7657 [info] WoR-PE (version 1.0.0.0)
2021-09-04 04:07:53.9911 [info] Copyright © 2018-2021 Mario Bălănică. All Rights Reserved.
2021-09-04 04:07:54.0278 [info] OS: Windows (TM) 10 Preinstallation Environment ver 2009 (22454.1000.arm64fre.rs_prerelease.210903-1516)
2021-09-04 04:07:54.1288 [info] Loading driver list...
2021-09-04 04:07:54.2450 [info] Loading driver (1/3): mcci_dwchsotg\mcci_dwchsotg_hcd.inf
2021-09-04 04:07:54.3303 [debug] Launch pnputil.exe with the following arguments: /add-driver "X:\drivers\mcci_dwchsotg\mcci_dwchsotg_hcd.inf" /install
2021-09-04 04:07:56.1879 [debug] pnputil.exe process output:
2021-09-04 04:07:56.1879 [debug] Standard output:
Microsoft PnP Utility

Adding driver package:  mcci_dwchsotg_hcd.inf
Driver package added successfully.
Published Name:         oem0.inf
Driver package installed on device: ACPI\BCM2848\0

Total driver packages:  1
Added driver packages:  1

2021-09-04 04:07:56.1879 [debug] Exit code: 0
2021-09-04 04:07:56.2483 [info] Loading driver (2/3): mcci_dwchsotg\mcci_dwchsotg_hub.inf
2021-09-04 04:07:56.2483 [debug] Launch pnputil.exe with the following arguments: /add-driver "X:\drivers\mcci_dwchsotg\mcci_dwchsotg_hub.inf" /install
2021-09-04 04:07:57.3778 [debug] pnputil.exe process output:
2021-09-04 04:07:57.3778 [debug] Standard output:
Microsoft PnP Utility

Adding driver package:  mcci_dwchsotg_hub.inf
Driver package added successfully.
Published Name:         oem1.inf
Driver package installed on device: USB\VID_040E&PID_0100&McciDwcHsOtgHub\0

Total driver packages:  1
Added driver packages:  1

2021-09-04 04:07:57.3778 [debug] Exit code: 0
2021-09-04 04:07:57.4088 [info] Loading driver (3/3): rpisdhc\rpisdhc.inf
2021-09-04 04:07:57.4234 [debug] Launch pnputil.exe with the following arguments: /add-driver "X:\drivers\rpisdhc\rpisdhc.inf" /install
2021-09-04 04:07:58.5118 [debug] pnputil.exe process output:
2021-09-04 04:07:58.5118 [debug] Standard output:
Microsoft PnP Utility

Adding driver package:  rpisdhc.inf
Driver package added successfully.
Published Name:         oem2.inf
Driver package installed on device: ACPI\BCM2855\0

Total driver packages:  1
Added driver packages:  1

2021-09-04 04:07:58.5118 [debug] Exit code: 0
2021-09-04 04:07:58.5491 [info] Drivers loaded successfully!
2021-09-04 04:08:02.9663 [info] Loading disks...
2021-09-04 04:08:05.1935 [info] Disk 0 - Generic Mass-Storage USB Device - Removable Media - 29 GB
2021-09-04 04:08:05.1935 [info] 1 disk(s) found
2021-09-04 04:08:05.2152 [info] Finding PE disk...
2021-09-04 04:08:05.2541 [info] Mounting partitions on disk 0...
2021-09-04 04:08:05.2815 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 1 , assign letter=A , exit
2021-09-04 04:08:05.3381 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:08:48.2117 [debug] diskpart.exe process output:
2021-09-04 04:08:48.2117 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 1 is now the selected partition.

DiskPart successfully assigned the drive letter or mount point.

Leaving DiskPart...

2021-09-04 04:08:48.2117 [debug] Exit code: 0
2021-09-04 04:08:48.2635 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 2 , assign letter=B , exit
2021-09-04 04:08:48.2791 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:08:50.1308 [debug] diskpart.exe process output:
2021-09-04 04:08:50.1308 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 2 is now the selected partition.

DiskPart successfully assigned the drive letter or mount point.

Leaving DiskPart...

2021-09-04 04:08:50.1308 [debug] Exit code: 0
2021-09-04 04:08:50.1636 [info] Partitions successfully mounted!
2021-09-04 04:08:50.1712 [info]   Boot: A
2021-09-04 04:08:50.1879 [info]   Win: B
2021-09-04 04:08:50.2863 [info] Found PE disk at index 0
2021-09-04 04:08:50.3000 [info] Selected disk: Disk 0 - Generic Mass-Storage USB Device - Removable Media - 29 GB - Installer
2021-09-04 04:08:50.3537 [info] Finding image file...
2021-09-04 04:08:50.3976 [info] Found image file at: B:\install.wim
2021-09-04 04:08:50.3976 [info] Getting image editions...
2021-09-04 04:08:50.9173 [info] Edition 1: Windows 10 Professional ARM64 build 22454.1000
2021-09-04 04:08:50.9443 [info] Selected edition: Windows 10 Professional build 22454.1000
2021-09-04 04:09:10.0510 [info] -- Installation details --
2021-09-04 04:09:10.0649 [info] PeInstallDisk / DestinationDisk
2021-09-04 04:09:10.0649 [info]   Index: 0
2021-09-04 04:09:10.0777 [info]   Name: Generic Mass-Storage USB Device
2021-09-04 04:09:10.0777 [info]   Type: Removable Media
2021-09-04 04:09:10.0777 [info]   Size: 29
2021-09-04 04:09:10.0971 [info] Image
2021-09-04 04:09:10.0971 [info]   Path: B:\install.wim
2021-09-04 04:09:10.1091 [info]   Index: 1
2021-09-04 04:09:10.1091 [info]   Name: Windows 10 Professional
2021-09-04 04:09:10.1091 [info]   Arch: ARM64
2021-09-04 04:09:10.1289 [info]   Build: 22454.1000
2021-09-04 04:09:10.1289 [info] -- End of installation details --
2021-09-04 04:09:10.1412 [info] The installation process has started!
2021-09-04 04:09:10.1786 [info] Creating temporary partition...
2021-09-04 04:09:10.1908 [debug] Creating diskpart script with the following commands: select disk 0 , create partition primary , exit
2021-09-04 04:09:10.2191 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:09:12.0924 [debug] diskpart.exe process output:
2021-09-04 04:09:12.0924 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

DiskPart succeeded in creating the specified partition.

Leaving DiskPart...

2021-09-04 04:09:12.0934 [debug] Exit code: 0
2021-09-04 04:09:12.1572 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 3 , format fs=ntfs quick , exit
2021-09-04 04:09:12.1572 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:09:17.3099 [debug] diskpart.exe process output:
2021-09-04 04:09:17.3099 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 3 is now the selected partition.


    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
  100 percent completed
DiskPart successfully formatted the volume.

Leaving DiskPart...

2021-09-04 04:09:17.3099 [debug] Exit code: 0
2021-09-04 04:09:17.3451 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 3 , assign letter=C , exit
2021-09-04 04:09:17.3451 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:09:18.9715 [debug] diskpart.exe process output:
2021-09-04 04:09:18.9715 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 3 is now the selected partition.

DiskPart successfully assigned the drive letter or mount point.

Leaving DiskPart...

2021-09-04 04:09:18.9715 [debug] Exit code: 0
2021-09-04 04:09:19.0114 [info] Temporary partition created successfully! (letter: C)
2021-09-04 04:09:19.0183 [info] Copying image file to the temporary partition...
2021-09-04 04:18:03.9890 [info] The image file has been copied to the temporary partition!
2021-09-04 04:18:03.9890 [info] Copying boot files to the temporary partition...
2021-09-04 04:18:04.8212 [info] The boot files have been copied to the temporary partition!
2021-09-04 04:18:04.8295 [info] Deleting the initial PE partitions...
2021-09-04 04:18:04.8419 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 2 , delete partition override , exit
2021-09-04 04:18:04.8560 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:07.4865 [debug] diskpart.exe process output:
2021-09-04 04:18:07.4865 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 2 is now the selected partition.

DiskPart successfully deleted the selected partition.

Leaving DiskPart...

2021-09-04 04:18:07.4865 [debug] Exit code: 0
2021-09-04 04:18:07.5204 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 1 , delete partition override , exit
2021-09-04 04:18:07.5204 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:09.1641 [debug] diskpart.exe process output:
2021-09-04 04:18:09.1641 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 1 is now the selected partition.

DiskPart successfully deleted the selected partition.

Leaving DiskPart...

2021-09-04 04:18:09.1641 [debug] Exit code: 0
2021-09-04 04:18:09.2125 [info] The initial PE partiitions have been deleted!
2021-09-04 04:18:09.2379 [info] Creating the partitions...
2021-09-04 04:18:09.2473 [debug] Creating diskpart script with the following commands: select disk 0 , create partition primary size=128 , exit
2021-09-04 04:18:09.2620 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:10.9632 [debug] diskpart.exe process output:
2021-09-04 04:18:10.9632 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

DiskPart succeeded in creating the specified partition.

Leaving DiskPart...

2021-09-04 04:18:10.9632 [debug] Exit code: 0
2021-09-04 04:18:11.0199 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 1 , format fs=fat32 quick label=Boot , exit
2021-09-04 04:18:11.0199 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:15.6753 [debug] diskpart.exe process output:
2021-09-04 04:18:15.6753 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 1 is now the selected partition.


    0 percent completed
    0 percent completed
    0 percent completed
  100 percent completed
DiskPart successfully formatted the volume.

Leaving DiskPart...

2021-09-04 04:18:15.6753 [debug] Exit code: 0
2021-09-04 04:18:15.7240 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 1 , assign letter=A , exit
2021-09-04 04:18:15.7369 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:17.2740 [debug] diskpart.exe process output:
2021-09-04 04:18:17.2740 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 1 is now the selected partition.

DiskPart successfully assigned the drive letter or mount point.

Leaving DiskPart...

2021-09-04 04:18:17.2740 [debug] Exit code: 0
2021-09-04 04:18:17.3235 [info] Boot partition created! (letter: A)
2021-09-04 04:18:17.3235 [debug] Creating diskpart script with the following commands: select disk 0 , create partition msr size=16 , exit
2021-09-04 04:18:17.3439 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:18.8922 [debug] diskpart.exe process output:
2021-09-04 04:18:18.8922 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

DiskPart succeeded in creating the specified partition.

Leaving DiskPart...

2021-09-04 04:18:18.8922 [debug] Exit code: 0
2021-09-04 04:18:18.9466 [info] MSR partition created!
2021-09-04 04:18:18.9562 [debug] Creating diskpart script with the following commands: select disk 0 , create partition primary , exit
2021-09-04 04:18:18.9562 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:20.7033 [debug] diskpart.exe process output:
2021-09-04 04:18:20.7033 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

DiskPart succeeded in creating the specified partition.

Leaving DiskPart...

2021-09-04 04:18:20.7033 [debug] Exit code: 0
2021-09-04 04:18:20.7454 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 4 , format fs=ntfs quick label=Windows , exit
2021-09-04 04:18:20.7591 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:28.2374 [debug] diskpart.exe process output:
2021-09-04 04:18:28.2374 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 4 is now the selected partition.


    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
    0 percent completed
  100 percent completed
DiskPart successfully formatted the volume.

Leaving DiskPart...

2021-09-04 04:18:28.2374 [debug] Exit code: 0
2021-09-04 04:18:28.2834 [debug] Creating diskpart script with the following commands: select disk 0 , select partition 4 , assign letter=B , exit
2021-09-04 04:18:28.2957 [debug] Launch diskpart.exe with the following arguments: /s "X:\diskpart.script"
2021-09-04 04:18:29.9096 [debug] diskpart.exe process output:
2021-09-04 04:18:29.9096 [debug] Standard output:

Microsoft DiskPart version 10.0.22454.1000

Copyright (C) Microsoft Corporation.
On computer: MINWINPC

Disk 0 is now the selected disk.

Partition 4 is now the selected partition.

DiskPart successfully assigned the drive letter or mount point.

Leaving DiskPart...

2021-09-04 04:18:29.9096 [debug] Exit code: 0
2021-09-04 04:18:29.9608 [info] Windows partition created! (letter: B)
2021-09-04 04:18:29.9889 [info] Applying the WIM image to: B
2021-09-04 04:43:10.6482 [error] System.ComponentModel.Win32Exception (1450): Insufficient system resources exist to complete the requested service.
   at Microsoft.Wim.WimgApi.ApplyImage(WimHandle imageHandle, String path, WimApplyImageOptions options) in Microsoft.Wim.dll:token 0x6000020+0x3a
   at WoR_PE.Wrappers.WIMGAPIWrapper.<>c__DisplayClass6_0.<ApplyImage>b__0() in WoR-PE.dll:token 0x60001c1+0x52
   at System.Threading.Tasks.Task.InnerInvoke() in System.Private.CoreLib.dll:token 0x6002d62+0xf
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in System.Private.CoreLib.dll:token 0x60029bb+0x4c
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in System.Private.CoreLib.dll:token 0x6002d61+0x12e
--- End of stack trace from previous location ---
   at WoR_PE.Wrappers.WIMGAPIWrapper.ApplyImage(String imageFile, String applyPath, UInt32 imageIndex, ProgressCallback callback) in WoR-PE.dll:token 0x6000084+0x9a
   at WoR_PE.Setup.SetupProcess.ApplyImageTask() in WoR-PE.dll:token 0x60000e3+0xe8
   at WoR_PE.Setup.SetupProcess.DeploymentTask() in WoR-PE.dll:token 0x60000de+0x29b
   at WoR_PE.Setup.SetupProcess.Run(IWin32Window parent) in WoR-PE.dll:token 0x60000dc+0xa7
2021-09-04 04:43:11.1954 [warn] The installation could not be finished!
2021-09-04 04:50:17.8048 [info] Elapsed time: 00:41:07
```

**Comment by Botspot** [comment]
I forwarded this to the developer of the PE-based installer. He says he thinks it's due to low RAM and will look into fixing this.

**Comment by wzrd-1** [comment]
I ran into an out of memory fault in BCM2711 on Pi4, running Raspbian 11. It reset my X session during the Microsoft download phase, leaving me with everything closed at the window manager login screen.
It turned out to be caused by a previous experiment in allocating 512 to the GPU. Once I reduced it to 256, everything worked properly.
Of course, complicating the issue was, I was certain that I had reverted that change.
